source/simulator/helper_functions.cc

```diff
@@ -909,8 +909,166 @@ namespace aspect
 
     return (material_model->get_model_dependence().viscosity != MaterialModel::NonlinearDependence::none);
   }
-}
 
+
+
+  template <int dim>
+  void Simulator<dim>::apply_limiter_to_dg_solutions (const AdvectionField &advection_field)
+  {
+    AssertThrow (dim == 2,
+                 ExcMessage ("Only apply bound preserving limiter to the 2 dimensional problem"));
```

Comment:
Oh, that's a shame. What would it take to generalize this code to 3d?

Can you reformulate the error message to say "The bound preserving limiter is currently only implemented for 2-dimensional cases."?